wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -0,0 +1,1599 @@
+        try {
+            String message = request.getMessage();
+            GitUser committer = getUser();
+            if (message == null) {
+                throw new GitException("Message wasn't set");
+            }
+            if (committer == null) {
+                throw new GitException("Committer can't be null");
+            }
+            String committerName = committer.getName();
+            String committerEmail = committer.getEmail();
+            if (committerName == null || committerEmail == null) {
+                throw new GitException("Git user name and (or) email wasn't set", ErrorCodes.NO_COMMITTER_NAME_OR_EMAIL_DEFINED);
+            }
+            if (!repository.getRepositoryState().canCommit()) {
+                Revision rev = newDto(Revision.class);
+                rev.setMessage(String.format(MESSAGE_COMMIT_NOT_POSSIBLE, repository.getRepositoryState().getDescription()));
+                return rev;
+            }
+
+            if (request.isAmend() && !repository.getRepositoryState().canAmend()) {
+                Revision rev = newDto(Revision.class);
+                rev.setMessage(String.format(MESSAGE_AMEND_NOT_POSSIBLE, repository.getRepositoryState().getDescription()));
+                return rev;
+            }
+
+            Status status = status(StatusFormat.LONG);
+            if (status.getAdded().isEmpty() && status.getChanged().isEmpty() && status.getRemoved().isEmpty()) {
+                if (request.isAll()) {
+                    if (status.getMissing().isEmpty() && status.getModified().isEmpty()) {
+                        return newDto(Revision.class).withMessage(message);
+                    }
+                } else {
+                    if (status.getMissing().isEmpty() && status.getModified().isEmpty()) {
+                        return newDto(Revision.class).withMessage(message);
+                    } else {
+                        return newDto(Revision.class).withMessage(message);
+                    }
+                }
+            }
+
+            CommitCommand commitCommand = getGit().commit()
+                                                  .setCommitter(committerName, committerEmail).setAuthor(committerName, committerEmail)
+                                                  .setMessage(message)
+                                                  .setAll(request.isAll())
+                                                  .setAmend(request.isAmend());
+
+            RevCommit result = commitCommand.call();
+            GitUser gitUser = newDto(GitUser.class).withName(committerName).withEmail(committerEmail);
+
+            return newDto(Revision.class).withBranch(getCurrentBranch())
+                                         .withId(result.getId().getName()).withMessage(result.getFullMessage())
+                                         .withCommitTime((long)result.getCommitTime() * 1000).withCommitter(gitUser);
```

Comment:
about * 1000 https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html#toMillis(long) or  MILLISECONDS.convert(duration, this) ?